### DISCUSSION TITLE
Greetings from FOSDEM

### DISCUSSION THREAD

**Discussion by johanvdw** [pretalx/pretalx#1637] [General]
Hello pretalx,

This is a post to share some experiences of running pretalx for [FOSDEM](https://fosdem.org). For those who don't know FOSDEM, it is one of the largest open source conferences which takes places every year in Brussels, Belgium.

Anyway, after many years of pentabarf, we decided to switch to pretalx for submissions to our conference. This was not evident, as FOSDEM is quite a bit larger than the average conference, eg for this 2024 edition we currently have (and still counting)

* 1860 submissions
* 65 tracks
* 36 rooms
* \>4000 reviews (still happening for some tracks)
* approximately 800 talks, almost 400 scheduled now

Managing this in *any* conference management system is not easy, and we knew when we started out we needed some customization. 

When starting out three modules were created:

* an exporter for the tool (nanoc) that builds our [website](https://fosdem.org/2024/schedule/). We have a lot of custom tools built against the website so if we could replicate it's intermediate yaml format we could still generate the website as before.
* a [plugin](https://github.com/FOSDEM/pretalx-integration/tree/master/devroom-settings) providing extra settings per track. We couple a team to every track (track managers) which can review and schedule. They are also able to see all other talks because it is important to find talks which should be submitted in a different track or talks which were submitted to several tracks. Apart from that the track managers (we call them devroom managers) can set a few settings, eg if submission should be closed or whether an online qa should be available. Apart from that we have a second team of track reviewers. Devroom managers can invite people to that team themselves, offloading some work of the administrators. 
![image](https://github.com/pretalx/pretalx/assets/868434/b59b56b1-2d75-4ee4-bb0b-ce736b37498a)

* pentabarf had a nice audit log which was often used and we wanted to replicate this feature. While not perfect yet, I was able to create [something similar](https://github.com/FOSDEM/pretalx-integration/tree/master/pretalx_auditlog) using django pghistory. This module is actually quite nice and probably deserves a separate repo once some rough edges have been cleaned off. It helps finding out when unexpected changes occur, or eg if you want to bulk change a setting for 65 tracks you can do one in the interface and then check which model was changed and update the rest through sql or in the django shell. I actually think it can also be useful for developers. Eg I see a lot of fields have their updated time being updated when that is probably not needed.
![image](https://github.com/pretalx/pretalx/assets/868434/621008e2-6f15-4878-a0ec-4a9308fe4629)

I also make some changes to [pretalx itself](https://github.com/johanvdw/pretalx/tree/fosdem_2024) (not in a plugin). I want to limit these kind of changes, but sometimes it was the quickest way to customize things while preparing the conference.

* I added [custom signals](https://github.com/johanvdw/pretalx/commit/c05437f5b2cfe7ab6a9024de802e9f64b7e8c11a) to add our custom model with track settings to the track page. This is actually an interesting case where I'd like to hear recommendations from the team, because we will probably add fields to room and user in the future.
* we made sure track_mail (a field from a custom moel we made) can be used in signals, as a reply-to address to make sure devroom managers would get answers instead of the admins
* the schedule editor was updated to filter (quite dirty) per track and to hide all rooms by default. This way devroom managers are less prone to make mistakes (though we don't block them from changing anything else in the schedule, hence the audit log).
![image](https://github.com/pretalx/pretalx/assets/868434/a07f59bf-e73a-49e8-bcef-bfe9e9a4c539)
* releases will be automated, the release button was changed to make sure devroom managers can see the conflicts.

The good news: the system still holds, users are happy and pretalx made most of the changes quite easy. Hey I even found time to write this post rather than figuring out any of the issues :-)

**Comment by rixx** [comment]
Thank you *so* much for sharing your experiences with pretalx, and also for your regular feedback via issues and PRs! I kept feeling vaguely bad that you were running into so many niggles, so it's great to see that things work well overall (especially considering that migrating a system with this many moving parts is never easy)!

I'd be more than happy to work with you on things you need pretaly to do!

- Adding new signals is no trouble at all. I'd probably want to consider potential alternate ways of doing things in some cases – eg it might be better / more Django-like to offer a signal to add more form fields to the track form (or adding more forms …), rather than adding custom HTML. All up for debate, though, I just would like to have a concept that won't hurt to expand on in the future.
- A better audit log is something I've been wanting for a while now. I hope I'll find the time to look at django-pghistory and similar tools – the trouble with eg updated timestamp fields is that those are currently set via Django in the model `save` method. sigh – not ideal for sure!
- Automating releases (probably via a setting) is something I've toyed with for a while. We at the very least need a better indicator for unreleased changes (a feature I had hoped to stuff into the end-of-year release, but didn't find the time to work on).

I'm also eyeing some of your changes jealously – having some form of track/team management is something I've been wanting for a while, but as you've probably discovered, it doesn't fit in well with pretalx's UI and permission handling at the moment :/ Same with changing who gets/is reply-to for specific emails.

all in all: thank you for your help and your feedback. Always happy to chat about things pretalx could do better, either here in the issues or via email (or other text-based media, really). As you already know, I'm happy to review and merge contributions, too – and I'm also occasionally building commissioned features, so if you ever don't feel like building something FOSDEM needs, that'd also be an option to keep in mind.
